### PR TITLE
feat: implement communication with back-end for the chatbot

### PR DESCRIPTION
**Pull request by ahmtslmngcl**
- services/api/chatbot.ts: implemented the basic logic for back-end communication
- types/chatbot.ts: changed IMessage interface so that the format is according to openai api
- components/Chatbot/Chatbot.tsx: adjusted send message with response, message history is also included

IMPORTANT: the implementation needs also some adjustments on the back-end side. (I will make a pull request for those changes in a couple of minutes)

### CODE REVIEW

**Comment by Esoteriker** [comment]
Good job! LGTM!

File: src/components/Chatbot/Chatbot.tsx [hunk 1]
```diff
@@ -244,26 +251,26 @@ export default function HungerMapChatbot() {
...
                         <div
                           className={`${
-                            message.sender === 'user'
+                            message.role === 'user'
                               ? isDarkMode
                                 ? 'chat-user-message-dark'
                                 : 'chat-user-message-light'
                               : 'chat-ai-message'
-                          } rounded-lg p-3 max-w-[80%] ${message.sender === 'user' ? 'ml-12' : ''}`}
+                          } rounded-lg p-3 max-w-[80%] ${message.role === 'user' ? 'ml-12' : ''}`}
```
Comment (Tschonti) [inline src/components/Chatbot/Chatbot.tsx:268]: I know this is merged, but next time take a look at [clsx](https://www.npmjs.com/package/clsx)

File: src/services/api/chatbot.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  previous_messages?: IMessage[];
+}
+
+export interface QueryResponse {
+  response: string;
+  prompt_tokens: number;
+  total_tokens: number;
+}
+
+const API_BASE_URL = 'http://127.0.0.1:5000'; // Adjust this when back-end is deployed and in production
```
Comment (Tschonti) [inline src/services/api/chatbot.ts:21]: this should go to .env

**Comment by Tschonti** [comment]
next time please wait for a review from @marinovl7

**Comment by Tschonti** [comment]
> next time please wait for a review from @marinovl7

I see now this PR wasn't into main, then it's fine, sorry

File: src/components/Chatbot/Chatbot.tsx (hunk 1 above)
Comment (Esoteriker) [reply src/components/Chatbot/Chatbot.tsx:268]: Nice tip, I didn't know clsx before!

**Pull request merged**